account_bank_receipt/models/account_bank_receipt.py

```diff
@@ -0,0 +1,378 @@
+        }
+
+    def validate_deposit(self):
+        am_obj = self.env['account.move']
+        move_line_obj = self.env['account.move.line']
+        for deposit in self:
+            move_vals = self._prepare_account_move_vals(deposit)
+            move = am_obj.create(move_vals)
+            total_amount = 0.0
+            total_amount_currency = 0.0
+            rate_currency = False
+            to_reconcile_lines = []
+            if deposit.company_id.currency_id != deposit.currency_id:
+                rate_currency = self.currency_id._get_conversion_rate(
+                    self.company_id.currency_id, self.currency_id,
+                    self.company_id, self.receipt_date)
+            for line in deposit.receipt_line:
+                if rate_currency:
+                    total_amount += line.allocation/rate_currency
+                    total_amount_currency += line.allocation
+                else:
+                    total_amount += line.allocation
+                line_vals = self._prepare_move_line_vals(line, rate_currency)
+                line_vals['move_id'] = move.id
+                move_line = move_line_obj.with_context(
+                    check_move_validity=False).create(line_vals)
+                to_reconcile_lines.append(line.move_line_ids + move_line)
+
+            # Create counter-part
+            counter_vals = self._prepare_counterpart_move_lines_vals(
+                deposit, total_amount, total_amount_currency)
+            counter_vals['move_id'] = move.id
+            move_line_obj.create(counter_vals)
+            if deposit.company_id.bank_receipt_post_move:
+                move.post()
+
+            deposit.write({'state': 'done', 'move_id': move.id})
+            for reconcile_lines in to_reconcile_lines:
+                reconcile_lines.reconcile()
+        return True
+
+    @api.onchange('company_id')
+    def onchange_company_id(self):
+        if self.company_id:
+            bank_journals = self.env['account.journal'].search([
+                ('company_id', '=', self.company_id.id),
+                ('type', '=', 'bank'),
+                ('bank_account_id', '!=', False)])
+            if len(bank_journals) == 1:
+                self.bank_journal_id = bank_journals[0]
+        else:
+            self.bank_journal_id = False
+
+    @api.onchange('journal_id')
+    def onchange_journal_id(self):
+        if self.journal_id:
+            if self.journal_id.currency_id:
+                self.currency_id = self.journal_id.currency_id
+            else:
+                self.currency_id = self.journal_id.company_id.currency_id
+
+
+class AccountBankReceiptLine(models.Model):
+    _name = "account.bank.receipt.line"
+    _description = "Account Bank Receipt Line"
+    _order = 'order_id, id'
+
+    order_id = fields.Many2one(
+        comodel_name='account.bank.receipt',
+        string='Order Reference',
+        copy=False,
+    )
+    move_line_ids = fields.Many2one(
```

Comment:
Many2one always use _id
```suggestion
    move_line_id = fields.Many2one(
```